actionview/lib/action_view/component.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module ActionView
+  class Component < ActionView::Base
+    include ActiveModel::Validations
+
+    attr_accessor :content
+
+    # As of now, we don't need anything in our base initializer.
+    # However, we don't want the `initialize` behavior from `ActionView::Base` right now.
+    def initialize(*)
+    end
+
+    # Entrypoint for rendering components. Called by ActionView::RenderingHelper#render.
+    #
+    # Returns ActionView::OutputBuffer.
+    def self.call(view_context, *args, &block)
+      @compiled ||= nil
+
+      compile unless @compiled
+      @compiled = true
+
+      instance = new(*args)
+      instance.content = view_context.capture(&block) if block_given?
```

Comment:
I'd like it if we could pass this in to `new` rather than using a setter.  I know it's to do with the superclass not matching signatures, but I wanted to comment with that anyway.  Using a setter like this will complicate downstream uses because they'll need to know to call the setter.  On the flip side, making it a required parameter to `initialize` will teach people it's required because they won't be able to construct the object without it.  Plus you don't even need to read the docs to figure that out! 😊

Reply:
Interesting. I'm not opposed to that approach, but I wonder how it might look for components that _don't_ render `content`. 

Right now they just ignore the `nil` `content` accessor, but if we were _always_ passing a `content` argument wouldn't they need to explicitly no-op that argument? Like so:

```ruby
def initialize(_content:); end
```